torch/cuda/__init__.py

```diff
@@ -305,7 +313,7 @@ def __enter__(self):
         self.prev_idx = torch.cuda._exchange_device(self.idx)
 
     def __exit__(self, type: Any, value: Any, traceback: Any):
-        self.idx = torch.cuda._exchange_device(self.prev_idx)
+        self.idx = torch.cuda._maybe_exchange_device(self.prev_idx)
```

Comment:
We should remove `_exchange_device` if we remove all the callsites for it!

Reply:
We still need `_exchange_device` at the entrance of this context manager.

Reply:
Ho right sorry.
DO you actually use self.idx here? This could be a maybe_set_device?

Reply:
You are right, `self.idx` is not used. Unfortunately, without assignment operation it was failing on torchscript #95305 #95306 , I am not sure if #96364 finally fixed it.